src/main/java/za/ac/sun/cs/search/adversarialagent/Board.java

```diff
@@ -0,0 +1,164 @@
+package za.ac.sun.cs.search.adversarialagent;
+
+import java.util.ArrayList;
+import java.lang.StringBuilder;
+
+import za.ac.sun.cs.search.adversarialagent.Move;
+
+public class Board {
+    private int m;
+    private int n;
+    private int k;
+    private int[][] board;
```

Comment:
We can probably make these `final`. Otherwise we might spend some time running after broken references again :fire:

Reply:
Will get on that.